### PR TITLE
log: generate digest and log it in slow log

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

follow up to https://github.com/pingcap/parser/pull/32, add digest to slow log

so we can awk or sql agg sql by digest to analysis query faster(even without pt-digest)

### What is changed and how it works?

- for each query generate digest and cache it in stmtCtx(later sql bind or audit can use it directly)
- add slow log filed
- add slow log table field 
- add `admin show slow` field(maybe we can do more use digest for show slow but maybe in following pr)

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Add stmt variable

Side effects

 - Possible performance regression(but lexer is fast)

Related changes

 - Need to cherry-pick to the release 2.1 when master work well

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/9662)

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -953,6 +953,8 @@ func (s *session) execute(ctx context.Context, sql string) (recordSets []sqlexec
 		}
 		metrics.SessionExecuteCompileDuration.WithLabelValues(label).Observe(time.Since(startTS).Seconds())
 
+		s.sessionVars.StmtCtx.Normalized, s.sessionVars.StmtCtx.SQLDigest = parser.NormalizeDigest(stmtNode.Text())
```
Comment (zz-jason) [inline session/session.go]: how about lazily compute the SQL digest when we need it?

Comment (lysu) [reply session/session.go]: good idea, done, PTAL thx~

**Comment by lysu** [comment]
/run-common-test tidb-test=pr/755
/run-integration-common-test=pr/755

**Pull request merged**